packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts

```diff
@@ -8,20 +8,55 @@
 import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
 import { ComponentSet } from '@salesforce/source-deploy-retrieve';
 import { WorkspaceContext } from '../../context/workspaceContext';
+import { SfdxProjectConfig } from '../../sfdxProject';
+
+export async function setApiVersion(componentSet: ComponentSet): Promise<void> {
+  // For a listing (and order of precedence) of how to retrieve the value of apiVersion,
+  // see "apiVersion: Order of Precedence" in the "How API Version and Source API Version
+  // Work in Salesforce CLI" doc.
+  // https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_apiversion.htm
 
-export async function setApiVersionOn(components: ComponentSet) {
   // Check the SFDX configuration to see if there is an overridden api version.
   // Project level local sfdx-config takes precedence over global sfdx-config at system level.
   const userConfiguredApiVersion = await ConfigUtil.getUserConfiguredApiVersion();
-
   if (userConfiguredApiVersion) {
-    components.apiVersion = userConfiguredApiVersion;
+    componentSet.apiVersion = userConfiguredApiVersion;
     return;
   }
 
   // If no user-configured Api Version is present, then get the version from the Org.
   const orgApiVersion = await getOrgApiVersion();
-  components.apiVersion = orgApiVersion ?? components.apiVersion;
+  componentSet.apiVersion = orgApiVersion ?? componentSet.apiVersion;
+}
+
+export async function setSourceApiVersion(componentSet: ComponentSet): Promise<void> {
```

Comment:
And then this is our new guy - I hear you in the comments being helpful! Can you add tests for this one? It looks like we only have coverage for setApiVersion.